config/initializers/default_url_options.rb

```diff
@@ -0,0 +1,4 @@
+uri = URI(HostingEnvironment.application_url)
+Rails.application.routes.default_url_options[:host] = uri.host
+Rails.application.routes.default_url_options[:port] = uri.port
+Rails.application.routes.default_url_options[:protocol] = uri.scheme
```

Comment:
What is this for?

Reply:
Configures default_url_options across the entire app, so that url helpers (as opposed to path helpers) use the right domain, port and protocol, according to HostingEnvironment.